### PR TITLE
Add Ruby 3.4

### PR DESCRIPTION
**Pull request by bslobodin**
Adds support for Ruby 3.4 going forward.

### CODE REVIEW

File: lib/money/splitter.rb [hunk 1]
```diff
@@ -77,7 +77,6 @@ def [](index)
           return money
         end
       end
-      nil
```
Comment (bslobodin) [inline lib/money/splitter.rb:80]: image

Comment (elfassy) [reply lib/money/splitter.rb:80]: Screenshot 2025-04-11 at 9 04 56 AM

CI runs simplecov as and checks that we have 100% coverage, strange that it didn't pick up on it 🤔 Are you sure you ran all the tests?

Comment (elfassy) [reply lib/money/splitter.rb:80]: yes I think we're missing a test for the case where this returns nil. Strange that this didn't make CI fail before ruby 3.4

**Review by elfassy — APPROVED** [review]

File: lib/money/splitter.rb (hunk 1 above)
Comment (elfassy) [reply lib/money/splitter.rb]: `the last expression would be split.each do ... end. The each method on a Hash returns the hash itself`
